crates/codegen/syntax_templates/src/shared/parse_error.rs

```diff
@@ -0,0 +1,78 @@
+use std::ops::Range;
+
+use super::{kinds::*, parse_output::ParseError};
+
+impl ParseError {
+    #[allow(dead_code)]
+    pub(crate) fn new_at_position(
+        utf8_position: usize,
+        tokens_that_would_have_allowed_more_progress: Vec<TokenKind>,
+    ) -> Self {
+        Self {
+            utf8_range: Range {
+                start: utf8_position,
+                end: utf8_position,
+            },
```

Comment:
suggestion for these places and others, using the shorter syntax for compactness:

```suggestion
            utf8_range: utf8_position..utf8_position,
```

Reply:
done